src/main/java/crazypants/enderio/machine/farm/TileFarmStation.java

```diff
@@ -321,21 +347,21 @@ protected boolean isMachineItemValidForSlot(int i, ItemStack stack) {
     }
     if(i <= maxToolSlot) {
       if (ToolType.isTool(stack)) {
-        int otherSlot = i == minToolSlot ? maxToolSlot : minToolSlot;
-        if (inventory[otherSlot] == null) {
-          return true;
-        } else { // let's make sure there's not one of this type in here already
-          for (ToolType type : ToolType.values()) {
-            if (type.itemMatches(inventory[otherSlot]) && type.itemMatches(stack)) {
-              return false;
-            }
+        for (int j = minToolSlot; j <= maxToolSlot; j++) {
+          if (ToolType.getToolType(stack).itemMatches(inventory[j])) {
+            return false;
           }
-          return true;
         }
+        return true;
       }
       return false;
+    } else if (i <= maxFirtSlot) {
+      return stack.getItem() == Items.dye && stack.getItemDamage() == 15;
```

Comment:
Short Google check says that MFRs Industrial Fertilizer does not support right-clicking by hand. So it'd need special coding anyway.Forestry also looks like it needs machine application. IC2 might work directly.